modules/plugin-search.php

```diff
@@ -0,0 +1,486 @@
+				$overrides = array(
+					'plugin-search' => true, // Helps to determine if that an injected card.
+					'name' => sprintf(       // Supplement name/description so that they clearly indicate this was added.
+						esc_html_x( 'Jetpack: %s', 'Jetpack: Module Name', 'jetpack' ),
+						$jetpack_modules_list[ $matching_module ]['name']
+					),
+					'short_description' => $jetpack_modules_list[ $matching_module ]['short_description'],
+					'requires_connection' => (bool) $jetpack_modules_list[ $matching_module ]['requires_connection'],
+					'slug'    => self::$slug,
+					'version' => JETPACK__VERSION,
+					'icons' => array(
+						'1x'  => "$image_url-128.png",
+						'2x'  => "$image_url-256.png",
+						'svg' => "$image_url.svg",
+					),
+				);
+
+				// Splice in the base module data
+				$inject = array_merge( $inject, $jetpack_modules_list[ $matching_module ], $overrides );
+
+				// Add it to the top of the list
+				array_unshift( $result->plugins, $inject );
+			}
+		}
+		return $result;
+	}
+
+	/**
+	 * Take a raw search query and return something a bit more standardized and
+	 * easy to work with.
+	 *
+	 * @param  String $term The raw search term
+	 * @return String A simplified/sanitized version.
+	 */
+	private function sanitize_search_term( $term ) {
+		$term = strtolower( urldecode( $term ) );
+
+		// remove non-alpha/space chars.
+		$term = preg_replace( '/[^a-z ]/', '', $term );
+
+		// remove strings that don't help matches.
+		$term = trim( str_replace( array( 'jetpack', 'jp', 'free', 'wordpress' ), '', $term ) );
+
+		return $term;
+	}
+
+	/**
+	 * Callback function to sort the array of modules by the sort option.
+	 */
+	private function by_sorting_option( $m1, $m2 ) {
+		return $m1['sort'] - $m2['sort'];
+	}
+
+	/**
+	 * Builds a URL to purchase and upgrade inserting the site fragment and the affiliate code if it exists.
+	 *
+	 * @param string $feature Module slug (or forged one for extra features).
+	 *
+	 * @since 7.1.0
+	 *
+	 * @return string URL to upgrade.
+	 */
+	private function get_upgrade_url( $feature ) {
+		$site_raw_url = Jetpack::build_raw_urls( get_home_url() );
+		$affiliateCode = Jetpack_Affiliate::init()->get_affiliate_code();
+		$user = wp_get_current_user()->ID;
+		return "https://jetpack.com/redirect/?source=plugin-hint-upgrade-$feature&site=$site_raw_url&u=$user" .
+		       ( $affiliateCode ? "&aff=$affiliateCode" : '' );
+	}
+
+	/**
+	 * Modify the URL to the feature settings, for example Publicize.
+	 * Sharing is included here because while we still have a page in WP Admin,
+	 * we prefer to send users to Calypso.
+	 *
+	 * @param string $feature
+	 *
+	 * @since 7.1.0
+	 *
+	 * @return string
+	 */
+	private function get_configure_url( $feature ) {
+		$url = Jetpack::module_configuration_url( $feature );
+		$siteFragment = Jetpack::build_raw_urls( get_home_url() );
+		switch ( $feature ) {
+			case 'sharing':
+			case 'publicize':
```

Comment:
I don't think that worked in my tests. The "configure" link for Publicize was still the link to the Jetpack Settings screen.